src/R/Components/Impl/ConnectionManager/Implementation/Connection.cs

```diff
@@ -20,5 +30,43 @@ internal class Connection : ConnectionInfo, IConnection {
         /// If true, the connection is to a remote machine
         /// </summary>
         public bool IsRemote { get; }
+
+        public override string UserProvidedPath {
+            get { return base.UserProvidedPath; }
+            set {
+                base.UserProvidedPath = value;
+                Path = ToCompletePath(value);
+            }
+        }
+
+        public static string ToCompletePath(string path) {
```

Comment:
The other option that I would be fine with, is treatment similar to how Chrome handles URLs, in that we do a _two-way_ conversion. In other words, we just say that the single canonical form for an URL with `https://` and default port number is to omit both. So if you type `foo`, we keep it as `foo` in UI, but treat it as `https://foo:5000` when connecting. But if you type `https://foo:5000`, we strip it down to just `foo`. For example, try typing "http://example.com:80" into Chrome address bar, and see what it does.